snapcraft/internal/grammar/errors.py

```diff
@@ -17,15 +17,20 @@
 from snapcraft.internal import errors
 
 
-class StagePackageSyntaxError(errors.SnapcraftError):
+class GrammarError(errors.SnapcraftError):
+    """Base class for grammar-related errors."""
+    pass
 
-    fmt = 'Invalid syntax for stage packages: {message}'
+
+class GrammarSyntaxError(GrammarError):
```

Comment:
That reads a little funny. Makes me wonder if it's a syntax error or wrong grammar ;-)

Generally the name grammar makes sense to me since we want to use it in more places and add new selectors.